docs/introduction/reducers.md

```diff
@@ -0,0 +1,87 @@
+!!! note
+	This section assumes you are already familiar with [Actions](actions.md) and [The Store](store.md).
+
+When you initialize your `store` with [`Store.new`](../api-reference.md#storenew), you provide a single function called a `reducer` which will consume any `action` dispatched to your `store` and create a new `state` object tree based on the current `state` object tree of your `store`.
+
+```lua
+local phoneNumberReducer = function(state, action)
+	if action.type == "ReceivedNewPhoneNumber" then
+		return action.phoneNumber
+	end
+
+	return state
+end
+```
+
+Note that `state` is never actually modified by our `reducer`. The `state` object tree of our `store` is *read-only*, so our `reducer` must construct a new `state` in response to the received `action`.
+
+For complex applications, it is often useful to break down the global `reducer` you provide to the `store` into a set of smaller `reducer` functions, each of which is responsible for a portion of the `state` object tree.
+
+```lua
+local friendsReducer = function(state, action)
+	--[[
+		The state will be nil the first time this reducer is executed.
```

Comment:
The state MIGHT be nil, but if the user provides an initial state at store creation then it may not be.